cg_openmm/simulation/physical_validation.py

```diff
@@ -0,0 +1,282 @@
+    reporter = MultiStateReporter(output_data, open_mode="r")
+    analyzer = ReplicaExchangeAnalyzer(reporter)
+
+    states = reporter.read_thermodynamic_states()[0]
+    temperature_list = []
+    for s in states:
+        temperature_list.append(s.temperature)
+    
+    (
+        replica_energies,
+        unsampled_state_energies,
+        neighborhoods,
+        replica_state_indices,
+    ) = analyzer.read_energies()
+
+    n_particles = np.shape(reporter.read_sampler_states(iteration=0)[0].positions)[0]
+    T_unit = temperature_list[0].unit
+    temps = np.array([temp.value_in_unit(T_unit) for temp in temperature_list])
+    beta_k = 1 / (kB.value_in_unit(unit.kilojoule_per_mole/T_unit) * temps)
+    n_replicas = len(temperature_list)
+    for k in range(n_replicas):
+        replica_energies[:, k, :] *= beta_k[k] ** (-1)
+
+    total_steps = len(replica_energies[0][0])
+    state_energies = np.zeros([n_replicas, total_steps])
+
+    for step in range(total_steps):
+        for state in range(n_replicas):
+            state_energies[state, step] = replica_energies[
+                np.where(replica_state_indices[:, step] == state)[0], 0, step
+            ]
+
+    state_energies *= unit.kilojoule_per_mole
+    
+    T_array = np.zeros(len(temperature_list))
+    for i in range(len(temperature_list)):
+        T_array[i] = temperature_list[i].value_in_unit(T_unit)
+    
+    if pairs.lower() != 'single' and pairs.lower() != 'adjacent' and pairs.lower() != 'all':
+        print(f"Error: Pair option '{pairs}' not recognized, using default option 'single'")
+        pairs = 'single'
+
+    if pairs.lower() == 'single': 
+    # Run ensemble validation on one optimally spaced temperature pair
+    
+        # Find optimal state pair for ensemble check:
+        # Compute standard deviations of each energy distribution:
+        state_energies_std = np.std(state_energies,axis=1)
+
+        # Select reference state point
+        T_ref = temperature_list[ref_state_index]
+        std_ref = state_energies_std[ref_state_index]
+
+        # Compute optimal spacing:
+        deltaT = 2*kB*T_ref**2/std_ref
+        #print("DeltaT: %r" %deltaT) 
+
+        # Find closest match
+        T_diff = np.abs(T_ref.value_in_unit(T_unit)-T_array)
+
+        T_opt_index = np.argmin(np.abs(deltaT.value_in_unit(T_unit) - T_diff))
+        T_opt = temperature_list[T_opt_index]
+
+        # Set SimulationData for physical validation
+
+        # State 1
+        state1_index = ref_state_index
+
+        sim_data1 = SimulationData()
+
+        sim_data1.observables = ObservableData(
+            potential_energy=state_energies[state1_index,:],
+            )
+            
+        sim_data1.ensemble = EnsembleData(
+            ensemble='NVT',
+            energy=state_energies[state1_index,:],
+            temperature=T_array[state1_index]
+            )
+            
+        sim_data1.units = UnitData(

```

Comment:
Can you just define a default UnitData object, then just state that sim_data1.units = default_UnitData?  Seems like that would reduce duplication / chance for errors - since they are all look like they are the same.